### ISSUE TITLE
Exception Checking Bug Fixes

### ISSUE THREAD

**Issue by patrickw276** [python/mypy#7514]
I've found two bugs in the logic for checking `raise` statements.

The first is incorrectly allowing values of `Type[Tuple]` when type checking Python 2 code:

```Python
x = (1, 2, 3)
t = type(x)
raise t
```

this type checks ok with the `--py2` flag but raises a type error at runtime.

The second is incorrectly reporting an error when raising a value with the type `Union[BaseException, Type[BaseException]`:

```Python
from typing import Type, Union
x: Union[BaseException, Type[BaseException]] = Exception
raise x
```

this fails type checking but is OK at runtime.

I have a PR for proposed changes to fix both these bugs. I believe all tests pass with these changes but they remove two conditional branches that I don't entirely understand. I'm a little worried I broke some untested case.
